src/jdk.incubator.code/share/classes/jdk/incubator/code/runtime/ReflectableLambdaMetafactory.java

```diff
@@ -0,0 +1,268 @@
+package jdk.incubator.code.runtime;
+
+import jdk.incubator.code.Op;
+import jdk.incubator.code.Quoted;
+import jdk.incubator.code.dialect.core.CoreOp.FuncOp;
+import jdk.internal.access.JavaLangInvokeAccess;
+import jdk.internal.access.JavaLangInvokeAccess.ReflectableLambdaInfo;
+import jdk.internal.access.SharedSecrets;
+
+import java.lang.constant.ClassDesc;
+import java.lang.constant.MethodHandleDesc;
+import java.lang.invoke.CallSite;
+import java.lang.invoke.LambdaConversionException;
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.SerializedLambda;
+
+public class ReflectableLambdaMetafactory {
+
+    private ReflectableLambdaMetafactory() {
+        // nope
+    }
+
+    /**
+     * Facilitates the creation of simple "function objects" that implement one
```

Comment:
For now I've just copied the javadoc of the original metafactory. If you want to go a different way, please shout. Perhaps a better way could be to say more succinctly that this works in the same way as the lambda metafactory but XYZ... (e.g. focus on the differences, rather than on the similarities)

Reply:
Yes, that would be more succinct. We can revisit after integration of this PR.